### DISCUSSION TITLE
Current state of compatibility with sparse-checkout / scalar

### DISCUSSION THREAD

**Discussion by ctanis** [git-lfs/git-lfs#5643] [Q&A] [answered]
We are trying to switch a gigantic legacy code to Git on Azure, using Scalar with sparse-checkout and LFS.
In all my experiments, a fresh clone using Scalar's cone-based partial clone will immediately lead to errors when running things like `git lfs fsck` and `git lfs ls-files`...    I can provide details of some errors, but in my most recent experiments at least all the errors  surround "missing objects" which are in .promisor files and can be fetched manually with no problem.

These errors do make sense, given my understanding of how sparse-checkout and LFS work, but what surprises me is how few questions there are about these errors.

Given this background, my question is "What is the current state of LFS combined with sparse-checkout"?   Are these errors to be expected and are relatively inert?  Do I have some kind of repository problem?  Note that if I disable sparse checkout and pull the whole repository there are no issues with `fsck` or `lfs fsck`.

I am using the latest Git for Windows with Scalar and LFS installed for most of these issues, but thankfully I get to spend more time in Linux and the same errors appear there, though the version of these tools is not the latest.

Thanks,
Craig

**Comment by bk2204** [comment, marked as answer]
Hey,

As far as the legacy VFS for Git protocol in Scalar, that's not supported.  As for sparse-checkout, the answer is that we don't know how well it does or doesn't work because it's untested with Git LFS.  There's a couple of issues tracking improving our support with sparse checkout, but it's not anything that anyone has spent a dedicated amount of time implementing or testing.

We'll happily accept patches to improve the situation, but at the moment I would not expect it to work, although it may in some cases.

**Reply by ctanis** [reply]
OK this is great info -- can you confirm whether you're making a distinction between the original  VFS filesystem support for legacy scalar, or the GVFS protocol still supported in the current microsoft fork of git's scalar?
I assume the answer is that you are not, but all of these options are bewildering enough that I figured it wouldn't help to ask.

**Reply by bk2204** [reply]
We have no intention of supporting any VFS-like protocol (old, new, or otherwise) related to Scalar because the protocol doesn't handle Git LFS and it's not seeing substantial enough upstream development to warrant integrating with it.  (In fact, it's my understanding that it continues to exist for legacy reasons only.)  The work to integrate with it would be extremely involved and difficult and we simply don't have the capacity to support that or maintain it, especially since there's only one forge that might support it at all.

There was also discussion of a VFS-related protocol that Google was developing on the Git list, and while I haven't discussed it with the rest of the core team, my position there is the same at this moment for many of the same reasons.

We will happily accept improved support for sparse checkout.
